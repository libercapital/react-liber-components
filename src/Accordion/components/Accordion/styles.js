import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
