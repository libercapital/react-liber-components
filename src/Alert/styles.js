import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
