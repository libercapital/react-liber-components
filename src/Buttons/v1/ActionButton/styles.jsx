import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
