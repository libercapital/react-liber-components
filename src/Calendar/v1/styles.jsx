import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../../Themes';

import { CardContent } from '../../Card';
import { FaAngleLeft, FaAngleRight } from '../../Icons';
import { Button } from '../../Buttons';
import { media } from '../../Util';

export const Dot = styled.span`
  width: 6px;
  height: 6px;
  background-color: ${props => handleThemeFromObject(props.theme, 'calendar.dot', '#cecece', props.color)};
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  display: inline-block;
  border-radius: 3px;
  margin-right: 8px;
`;

export const BiggerDot = styled(Dot)`
  width: unset;
  height: 18px;
  width: unset;
  min-width: 18px;
  border-radius: 9px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.biggerDot', '#fff')};
  margin-right: 0px;
  margin-bottom: 5px;
  font-weight: bold;
  padding: 0 5px;
  overflow: hidden;
  box-sizing: border-box;
`;

export const FloatingDots = styled.div`
  position: absolute;
  top: 4px;
  left: ${props => (props.today ? '32px' : '27px')};
  width: 18px;
  display: flex;
  overflow: hidden;
  flex-wrap: wrap;
  & ${Dot} {
    margin-bottom: 2px;
    margin-right: 2px;
  }
  ${media.small`
    left: ${props => (props.today ? '36px' : '31px')};
  `}
  ${media.medium`
    left: ${props => (props.today ? '36px' : '31px')};
  `}
  ${media.large`
    left: ${props => (props.today ? '40px' : '35px')};
  `}
  ${media.extraLarge`
    left: ${props => (props.today ? '40px' : '35px')};
  `}
`;

export const DataContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  font-family: inherit;
  font-size: 11px;
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.dataContainer', '#717171')};
  text-transform: none;
  margin-bottom: 12px;
  width: unset;
  ${media.small`
    font-size: 11px;
  `}
  ${media.medium`
    font-size: 11px;
  `}
  ${media.large`
    width: 65px;
    font-size: 13px;
  `}
  ${media.extraLarge`
    width: 65px;
    font-size: 13px;
  `}
`;

const tooltipPosition = (position) => {
  switch (position) {
    case 'right':
      return css`
        bottom: 30px;
        left: 30px;
      `;
    case 'left':
      return css`
        bottom: 30px;
        right: 30px;
      `;
    case 'middle':
    default:
      return css`
        bottom: 40px;
        left: calc(50% - calc(148px / 2));
      `;
  }
};

export const Tooltip = styled.div`
  position: absolute;
  ${props => tooltipPosition(props.position)}
  font-family: inherit;
  font-size: 12px;
  font-weight: 500;
  width: 148px;
  border-radius: 4px;
  background-color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.background', '#ffffff')};
  box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.2);
  color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.color', '#717171')};
  padding: 8px;
  display: ${props => (props.hide ? 'none!important' : 'none')};
  z-index: 1000;
  box-sizing: border-box;
  & > h6 {
    margin: 0px;
    opacity: 0.75;
    font-family: inherit;
    font-size: 12px;
    font-weight: bold;
    text-align: left;
    color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.color', '#717171')};
    text-transform: capitalize;
    text-align: center;
  }
  & > div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
`;

export const Day = styled.div`
  position: relative;
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.date.today.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.date.today.fontSize', '14px')};
  width: 28px;
  height: 28px;
  border: 0px;
  margin: 0px;
  padding: 0px;
  border-radius: 20px;
  user-select: none;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};

  background: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.background', 'linear-gradient(to bottom, #00ccff, #009dff)')
    : 'transparent'
  )};
  box-shadow:  ${props => (props.today ? '0 2px 6px 0 rgba(0, 0, 0, 0.05)' : null)};
  font-weight: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightToday', '700')
    : handleThemeFromObject(props.theme, 'calendar.date.today.fontWeight', '400')
  )};

  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.today.color', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.default', '#717171');
  }};

  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    background-color: rgba(230, 230, 230, 0.5);
    font-weight: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightToday', '700')
    : handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightHighlight', '500')
  )};
  }
  &:hover > ${Tooltip} {
    display: block;
  }
  &:hover ${FloatingDots} {
    display: ${props => (props.showDot ? null : 'none')};
  }
  ${media.small`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.large`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
`;

export const DayLink = styled.a`
  text-decoration: none;
  position: relative;
  font-family: inherit;
  font-size: 14px;
  width: 28px;
  height: 28px;
  border: 0px;
  margin: 0px;
  padding: 0px;
  border-radius: 20px;
  user-select: none;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};

  background: ${props => (props.today
    ? handleThemeFromObject(
      props.theme,
      'calendar.date.link.background',
      'linear-gradient(to bottom, #00ccff, #009dff)',
    )
    : 'transparent'
  )};
  box-shadow:  ${props => (props.today ? '0 2px 6px 0 rgba(0, 0, 0, 0.05)' : null)};
  font-weight: ${props => (props.today ? 'bold' : '500')};

  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.today', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.link.default', '#717171');
  }};

  display: flex;
  justify-content: center;
  align-items: center;


  &:hover {
    background-color: rgba(230, 230, 230, 0.5);
    color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.today', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.link.default', '#717171');
  }};
  }
  &:hover > ${Tooltip} {
    display: block;
  }
  &:hover ${FloatingDots} {
    display: ${props => (props.showDot ? null : 'none')};
  }
  ${media.small`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.medium`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.large`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
  ${media.extraLarge`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
`;

export const WeekDay = styled.div`
  width: 30px;
  height: 24px;
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSize', '12px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontWeight', '700')};
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.week.color', '#009dff')};
  text-transform: uppercase;

  ${media.small`
    width: 35.1px;
    font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSizeSmall', '14px')};
  `}
  ${media.large`
    width: 54.1px;
    font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSizeLarge', '20px')};
  `}
`;

export const Month = styled.div`
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontSize', '20px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontWeight', '500')};
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.month,color', '#717171')};
  text-transform: uppercase;
  width: unset;
  flex-grow: 1;
  ${media.large`
    width: 260px;
    flex-grow: 0;
  `}
`;

export const Week = styled(Month)`
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontSize', '20px')};
  width: unset;
  flex-grow: 1;

  ${media.large`
    width: 250px;
    flex-grow: 0;
  `}
`;

export const Row = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  min-height: 36px;
  max-height: 36px;
  ${media.small`
    min-height: 40px;
    max-height: 40px;
  `}
  ${media.medium`
    min-height: 40px;
    max-height: 40px;
  `}
  ${media.large`
    min-height: 43px;
    max-height: 43px;
  `}
  ${media.extraLarge`
    min-height: 43px;
    max-height: 43px;
  `}
`;

export const FilterRow = styled(Row)`
  justify-content: flex-start;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.filterRow', '#717171')} !important;
  font-size: 14px;
  font-weight: 500;
  align-items: flex-start;
  min-height: unset;
  max-height: unset;
  ${media.large`
    align-items: center;
  `}
  ${media.extraLarge`
    align-items: center;
  `}
`;

export const FiltersBox = styled(FilterRow)`
  flex-wrap: wrap;
  font-size: 10px;
  & > *:not(:last-child) {
    margin-bottom: 10px;
  }
  ${media.large`
    font-size: 14px;
    flex-wrap: nowrap;
    & > * {
      margin-bottom: 0px !important;
    }
  `}
  ${media.extraLarge`
    font-size: 14px;
    flex-wrap: nowrap;
    & > * {
      margin-bottom: 0px !important;
    }
  `}
`;

export const FilterLabel = styled.div`
  font-family: inherit;
  font-size: 14px;
  font-weight: 900;
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.filterLabel', '#717171')};
  margin-right: 20px;
`;

export const Cell = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const CalendarBody = styled(CardContent)`
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${props => (props.week ? '0px' : '10px')};
`;

export const Left = styled(FaAngleLeft)`
  width: 24px;
  height: 24px;
  fill: ${props => handleThemeFromObject(props.theme, 'calendar.left', '#009dff')};
`;

export const Right = styled(FaAngleRight)`
  width: 24px;
  height: 24px;
  fill: ${props => handleThemeFromObject(props.theme, 'calendar.right', '#009dff')};
`;

export const NavButton = styled(Button)`
  background-color: transparent;
  border-radius: 50%;
  padding: 0px;
  width: 40px;
  height: 40px;
  &:hover {
    background-color: ${props => (props.disabled ? null : 'rgba(230, 230, 230, 0.5)')};
  }
  margin: 0px;
  & > svg{
    fill: ${props => (props.disabled ? 'rgba(230, 230, 230, 0.5)' : null)};
  }
  ${media.small`
    margin: 0 15px;
  `}
  ${media.medium`
    margin: 0 15px;
  `}
  ${media.large`
    margin: 0 10px;
  `}
  ${media.extraLarge`
    margin: 0 10px;
  `}
`;

export const WeekBody = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: space-evenly;
  width: 100%;
`;

export const WeekDayValue = styled.div`
  display: inline-flex;
  flex-direction: column;
  font-family: inherit;
  font-size: 18px;
  width: 47px;
  font-weight: 500;
  margin-bottom: 12px;
  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.today', '#009dff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.default', '#717171');
  }};
  align-items: center;
  text-transform: uppercase;
  ${media.large`
    font-size: 20px;
  `}
  ${media.extraLarge`
    font-size: 20px;
  `}
`;

export const WeekCell = styled.div`
  flex-basis: 20%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  min-height: 300px;
  border-image:
    linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0)) 1 100%;
  &:last-child {
    border: 0px;
  }

  padding-top: 5px;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};
  border-top: 0px;
  border-left: 0px;
  &:hover {
    background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0));
  }
  &:hover > ${WeekDayValue} {
    font-weight: 900;
  }
  &:hover ${DataContainer} {
    font-weight: 900;
  }
`;

export const WeekCellLink = styled.a`
  text-decoration: none;
  flex-basis: 20%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  min-height: 300px;
  border-image:
    linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0)) 1 100%;
  &:last-child {
    border: 0px;
  }

  padding-top: 5px;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};
  border-top: 0px;
  border-left: 0px;
  &:hover {
    background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0));
  }
  &:hover > ${WeekDayValue} {
    font-weight: 900;
  }
  &:hover ${DataContainer} {
    font-weight: 900;
  }
`;

export const WeekNav = styled(Row)`
  padding: 15px 0px;
  border-bottom: 2px solid rgba(230, 230, 230, 0.3);
`;
