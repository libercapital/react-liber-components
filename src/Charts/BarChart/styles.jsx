import styled from 'styled-components';
import { handleThemeFromObject } from 'Themes';

export const empty = null;

export const Slider = styled.input`
  width: ${props => (props.width ? `${props.width}px` : '100%')};
  appearance: none;
  height: 8.5px;
  background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.background', '#fff')};
  outline: none;
  border-radius: 4.25px;
  border: solid 2px ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};

  &::-webkit-slider-thumb {
    -webkit-appearance: none;
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &::-moz-range-thumb {
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &::-ms-thumb {
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &:focus {
    outline: none;
  }
  &::-moz-focus-outer {
    border: 0;
  }
  &::-moz-range-track {
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.background', '#fff')};
  }
`;

Slider.defaultProps = {
  type: 'range',
  valueAsNumber: true,
};
