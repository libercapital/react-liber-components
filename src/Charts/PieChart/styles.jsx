import styled from 'styled-components';
import { Slice as SliceRaw, VictoryContainer } from 'victory';

export const Slice = styled(SliceRaw)`
  transition: all 300ms;
  cursor: pointer;
`;

export const PieBox = styled(VictoryContainer)`
  width: unset !important;
  height: unset !important;
  & > svg {
    width: ${props => (props.width ? `${props.width}px` : 'unset')} !important;
    height: ${props => (props.height ? `${props.height}px` : 'unset')} !important;
  }
`;
