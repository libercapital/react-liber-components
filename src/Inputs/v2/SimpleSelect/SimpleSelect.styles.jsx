import styled, { css } from 'styled-components';
import { handleThemeFromObject as getTheme } from '../../../Themes';
import { MdChevronDown } from '../../../Icons';

export const Container = styled.div`
  position: relative;
  margin-right: auto;
  width: 100%;
`;

export const DisplayBox = styled.div`
  width: 100%;
  height: 32px;
  border-radius: 4px;
  border: solid 1px #ecf0f3;
  background-color: ${props => getTheme(props.theme, 'colors.white', '#ffffff')};
  padding: 0px 12px;
  box-sizing: border-box;
  display: flex;
  align-items: center;
`;

export const ValueBox = styled.div`
  flex-grow: 1;
  line-height: 22px;
  font-size: ${props => getTheme(props.theme, 'font-sizes.p', '16px')};
  font-family: ${props => getTheme(props.theme, 'font-family', 'Roboto')};
  font-weight: ${props => getTheme(props.theme, 'font-weights.regular', '400')};
  color: ${props => getTheme(props.theme, 'colors.gray4', '#6287a7')};
  white-space: nowrap;
  max-width: calc(100% - 36px);
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const Chevron = styled(MdChevronDown)`
  --size: 18px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
  fill: ${props => getTheme(props.theme, 'colors.grey4', '#6287a7')};
  margin-left: 12px;
`;

const active = css`
  color: ${props => getTheme(props.theme, 'colors.petroleum', '#4b6f85')};
`;

const comom = css`
  color: ${props => getTheme(props.theme, 'colors.grey3', '#c6d1d8')};
`;

export const Option = styled.span`
  width: 100%;
  height: 40px;
  line-height: 22px;
  font-size: ${props => getTheme(props.theme, 'font-sizes.p', '16px')};
  font-family: ${props => getTheme(props.theme, 'font-family', 'Roboto')};
  font-weight: ${props => getTheme(props.theme, 'font-weights.medium', '500')};

  background-color: ${props => getTheme(props.theme, 'colors.white', '#ffffff')};
  display: flex;
  align-items: center;
  padding: 0 24px;
  cursor: pointer;

  ${props => (props.active ? active : comom)}

  &:hover {
    background-color: ${props => getTheme(props.theme, 'colors.grey7', '#f9fafa')};
  }
`;
