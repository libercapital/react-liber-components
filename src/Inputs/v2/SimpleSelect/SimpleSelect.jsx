import React, { Children, Component } from 'react';
import PropTypes from 'prop-types';
import innerText from 'react-innertext';
import { SimpleDropDown } from '../../../DropDown';
import { Container, DisplayBox, ValueBox, Chevron, Option } from './SimpleSelect.styles';

class SimpleSelect extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      label: '',
    };
  }

  static getDerivedStateFromProps(props, state) {
    if (props.value !== null) {
      const selected = Children.toArray(props.children).find(
        child => child.props.value === props.value,
      );
      if (selected) {
        return {
          value: props.value,
          label: innerText(selected),
        };
      }
    }
    return state;
  }

  makeSelection = (...args) => () => {
    const { onSelect } = this.props;
    if (this.dropdown) this.dropdown.close();
    onSelect(...args);
  };

  handleOptionSelection = (value, label) => () => {
    this.setState(
      {
        value,
        label,
      },
      this.makeSelection(value, label),
    );
  };

  mappingOption = currentValue => child => {
    const label = innerText(child);
    const value = child.props.value || label.toLowerCase();
    return (
      <Option
        key={value}
        {...child.props}
        label={label}
        value={value}
        active={value === currentValue}
        onClick={this.handleOptionSelection(value, label)}
      />
    );
  };

  dropDownRef = element => {
    this.dropdown = element;
  };

  render() {
    const { value, label } = this.state;
    const { children, ...props } = this.props;
    return (
      <Container>
        <SimpleDropDown
          ref={this.dropDownRef}
          customButton={
            <DisplayBox {...props}>
              <ValueBox>{label}</ValueBox>
              <Chevron />
            </DisplayBox>
          }
        >
          {Children.toArray(children).map(this.mappingOption(value))}
        </SimpleDropDown>
      </Container>
    );
  }
}

SimpleSelect.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]),
  value: PropTypes.string,
  onSelect: PropTypes.func,
};

SimpleSelect.defaultProps = {
  value: null,
  children: [],
  onSelect: () => {},
};

export default SimpleSelect;
