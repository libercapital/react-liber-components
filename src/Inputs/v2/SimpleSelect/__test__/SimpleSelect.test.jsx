import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import SimpleSelect from '../SimpleSelect';
import { Option } from '../SimpleSelect.styles';

describe('Testing SimpleSelect Component', () => {
  configure({ adapter: new Adapter() });
  let props;

  const options = ['10', '25', '50', '100'];

  beforeEach(() => {
    props = {
      children: options.map(option => (<option value={option}>{option}</option>)),
      onSelect: sinon.spy(),
    };
  });
  it('renders SimpleSelect correctly without props', () => {
    const wrapper = shallow(<SimpleSelect />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders SimpleSelect correctly with props', () => {
    const wrapper = shallow(<SimpleSelect {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders SimpleSelect correctly with value', () => {
    props.value = '50';
    const wrapper = shallow(<SimpleSelect {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should select value on Option Click', () => {
    const wrapper = shallow(<SimpleSelect {...props} />);
    const opt = wrapper.find(Option);
    opt.at(1).simulate('click');
    expect(wrapper.state('value')).toBe('25');
    expect(wrapper.state('label')).toBe('25');
    expect(props.onSelect.calledWith('25', '25')).toBeTruthy();
  });
  it('should select value on Option Click with value', () => {
    props.value = '100';
    const wrapper = shallow(<SimpleSelect {...props} />);
    const opt = wrapper.find(Option);
    opt.at(1).simulate('click');
    expect(props.onSelect.calledWith('25', '25')).toBeTruthy();
  });
});
