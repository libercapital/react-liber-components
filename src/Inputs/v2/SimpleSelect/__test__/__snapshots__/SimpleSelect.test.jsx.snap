// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SimpleSelect Component renders SimpleSelect correctly with props 1`] = `
<SimpleSelectstyles__Container>
  <SimpleDropDown
    autoPosition={true}
    containerSize={false}
    customButton={
      <ForwardRef(SimpleSelectstyles__DisplayBox)
        onSelect={[Function]}
        value={null}
      >
        <ForwardRef(SimpleSelectstyles__ValueBox)>
          
        </ForwardRef(SimpleSelectstyles__ValueBox)>
        <ForwardRef(SimpleSelectstyles__Chevron) />
      </ForwardRef(SimpleSelectstyles__DisplayBox)>
    }
    hasOutsideLabel={false}
    label=""
    open={null}
    position="bottom-left-down-right"
    window={[Window]}
  >
    <SimpleSelectstyles__Option
      active={false}
      key="10"
      label="10"
      onClick={[Function]}
      value="10"
    >
      10
    </SimpleSelectstyles__Option>
    <SimpleSelectstyles__Option
      active={false}
      key="25"
      label="25"
      onClick={[Function]}
      value="25"
    >
      25
    </SimpleSelectstyles__Option>
    <SimpleSelectstyles__Option
      active={false}
      key="50"
      label="50"
      onClick={[Function]}
      value="50"
    >
      50
    </SimpleSelectstyles__Option>
    <SimpleSelectstyles__Option
      active={false}
      key="100"
      label="100"
      onClick={[Function]}
      value="100"
    >
      100
    </SimpleSelectstyles__Option>
  </SimpleDropDown>
</SimpleSelectstyles__Container>
`;

exports[`Testing SimpleSelect Component renders SimpleSelect correctly with value 1`] = `
<SimpleSelectstyles__Container>
  <SimpleDropDown
    autoPosition={true}
    containerSize={false}
    customButton={
      <ForwardRef(SimpleSelectstyles__DisplayBox)
        onSelect={[Function]}
        value="50"
      >
        <ForwardRef(SimpleSelectstyles__ValueBox)>
          50
        </ForwardRef(SimpleSelectstyles__ValueBox)>
        <ForwardRef(SimpleSelectstyles__Chevron) />
      </ForwardRef(SimpleSelectstyles__DisplayBox)>
    }
    hasOutsideLabel={false}
    label=""
    open={null}
    position="bottom-left-down-right"
    window={[Window]}
  >
    <SimpleSelectstyles__Option
      active={false}
      key="10"
      label="10"
      onClick={[Function]}
      value="10"
    >
      10
    </SimpleSelectstyles__Option>
    <SimpleSelectstyles__Option
      active={false}
      key="25"
      label="25"
      onClick={[Function]}
      value="25"
    >
      25
    </SimpleSelectstyles__Option>
    <SimpleSelectstyles__Option
      active={true}
      key="50"
      label="50"
      onClick={[Function]}
      value="50"
    >
      50
    </SimpleSelectstyles__Option>
    <SimpleSelectstyles__Option
      active={false}
      key="100"
      label="100"
      onClick={[Function]}
      value="100"
    >
      100
    </SimpleSelectstyles__Option>
  </SimpleDropDown>
</SimpleSelectstyles__Container>
`;

exports[`Testing SimpleSelect Component renders SimpleSelect correctly without props 1`] = `
<SimpleSelectstyles__Container>
  <SimpleDropDown
    autoPosition={true}
    containerSize={false}
    customButton={
      <ForwardRef(SimpleSelectstyles__DisplayBox)
        onSelect={[Function]}
        value={null}
      >
        <ForwardRef(SimpleSelectstyles__ValueBox)>
          
        </ForwardRef(SimpleSelectstyles__ValueBox)>
        <ForwardRef(SimpleSelectstyles__Chevron) />
      </ForwardRef(SimpleSelectstyles__DisplayBox)>
    }
    hasOutsideLabel={false}
    label=""
    open={null}
    position="bottom-left-down-right"
    window={[Window]}
  />
</SimpleSelectstyles__Container>
`;
